include/ideep/operators/conv.hpp

```diff
@@ -1222,16 +1222,29 @@ struct convolution_forward
       const lowp_kind alowp_kind = u8s8,
       const engine& aengine = engine::cpu_engine()) {
     bool is_channels_last = src.get_desc().is_channels_last() || weights.get_desc().is_channels_last();
-    if (bias.is_empty()) {
-      do_prepare</*with_bias=*/false>(
-          param, src, weights, bias, dst_dims, dst, strides, dilates,
-          padding_l, padding_r, groups, src_scales, weights_scales, dst_scales,
-          zero_point_t(), zero_point_t(), is_channels_last, attr, aalgorithm, aprop_kind, alowp_kind, aengine);
+    bool is_fp32 = src_scales.empty() && weights_scales.empty() && dst_scales.empty();
+    if (is_fp32) {
+      if (bias.is_empty()) {
+        do_prepare</*with_bias=*/false>(
+            param, src, weights, bias, dst_dims, dst, strides, dilates,
+            padding_l, padding_r, groups, is_channels_last, attr, aalgorithm, aprop_kind, aengine);
+      } else {
+        do_prepare</*with_bias=*/true>(
+            param, src, weights, bias, dst_dims, dst, strides, dilates,
+            padding_l, padding_r, groups, is_channels_last, attr, aalgorithm, aprop_kind, aengine);
+      }
     } else {
-      do_prepare</*with_bias=*/true>(
-          param, src, weights, bias, dst_dims, dst, strides, dilates,
-          padding_l, padding_r, groups, src_scales, weights_scales, dst_scales,
-          zero_point_t(), zero_point_t(), is_channels_last, attr, aalgorithm, aprop_kind, alowp_kind, aengine);
```

Comment:
Any reason why we don't check fp32 or int8 inside `do_prepare` and has to complicate the caller with the checks?

Reply:
Because we have different versions of `do_prepare` for int8 and fp32. The caller should determine which to use. For correctness, it is OK to use int8 version of `do_prepare` for fp32 ops but perf is not so good.